tests/Microsoft.Identity.Test.Unit/PublicApiTests/TelemetryClientTests.cs

```diff
@@ -153,56 +326,72 @@ private void AssertLoggedTelemetry(AuthenticationResult authenticationResult, Ms
             Assert.AreEqual(authenticationResult.AuthenticationResultMetadata.DurationInHttpInMs, eventDetails.Properties[TelemetryConstants.DurationInHttp]);
             Assert.AreEqual(authenticationResult.AuthenticationResultMetadata.DurationInCacheInMs, eventDetails.Properties[TelemetryConstants.DurationInCache]);
             Assert.AreEqual(authenticationResult.AuthenticationResultMetadata.DurationTotalInMs, eventDetails.Properties[TelemetryConstants.Duration]);
-        }
+            Assert.AreEqual(Convert.ToInt64(assertionType), eventDetails.Properties[TelemetryConstants.AssertionType]);
+            Assert.AreEqual(Convert.ToInt64(tokenType), eventDetails.Properties[TelemetryConstants.TokenType]);
+            Assert.AreEqual(endpoint, eventDetails.Properties[TelemetryConstants.Endpoint]);
 
-        private void CreateApplication()
-        {
-            _cca = ConfidentialClientApplicationBuilder
-                .Create(TestConstants.ClientId)
-                .WithClientSecret(TestConstants.ClientSecret)
-                .WithHttpManager(_harness.HttpManager)
-                .WithExperimentalFeatures()
-                .WithTelemetryClient(_telemetryClient)
-                .BuildConcrete();
-
-            TokenCacheHelper.PopulateCache(_cca.UserTokenCacheInternal.Accessor);
-        }
-    }
-
-    internal class TelemetryClient : ITelemetryClient
-    {
-        public MsalTelemetryEventDetails TestTelemetryEventDetails { get; set; }
-        
-        public TelemetryClient(string clientId)
-        {
-            ClientId = clientId;
-        }
-
-        public string ClientId { get; set; }
-
-        public void Initialize()
-        {
-
-        }
-
-        public bool IsEnabled()
-        {
-            return true;
+            if (eventDetails.Properties.ContainsKey(TelemetryConstants.CacheUsed))
```

Comment:
Assert that the cacheUsed is null when it is not set.

Reply:
I set it to the default value in the parameter so it will test that case.